{
    "schema_version": "1.6.0",
    "id": "OESA-2022-1566",
    "modified": "2022-03-07T00:00:00Z",
    "published": "2022-03-07T00:00:00Z",
    "withdrawn": "",
    "aliases": null,
    "related": [
        "CVE-2022-0391",
        "CVE-2021-4189"
    ],
    "summary": "python3 security update",
    "details": "Python combines remarkable power with very clear syntax. It has modules, classes, exceptions, very high level dynamic data types, and dynamic typing. There are interfaces to many system calls and libraries, as well as to various windowing systems. New built-in modules are easily written in C or C++ (or other languages, depending on the chosen implementation). Python is also usable as an extension language for applications written in other languages that need easy-to-use scripting or automation interfaces.\r\n\r\n\r\n\r\nSecurity Fix(es):\r\n\r\nA flaw was found in Python, specifically within the urllib.parse module. This module helps break Uniform Resource Locator (URL) strings into components. The issue involves how the urlparse method does not sanitize input and allows characters like \u0026apos;\\r\u0026apos; and \u0026apos;\\n\u0026apos; in the URL path. This flaw allows an attacker to input a crafted URL, leading to injection attacks. This flaw affects Python versions prior to 3.10.0b1, 3.9.5, 3.8.11, 3.7.11 and 3.6.14.(CVE-2022-0391)\r\n\r\nAn attacker could exploit this vulnerability to set up a malicious FTP server that could trick an FTP client into connecting back to a given IP address and port. This can cause the FTP client to scan the port.(CVE-2021-4189)",
    "severity": null,
    "affected": [
        {
            "package": {
                "ecosystem": "openEuler-20.03-LTS-SP1",
                "name": "python3",
                "purl": "pkg:rpm/python3@3.7.9-20.oe1"
            },
            "severity": null,
            "ranges": [
                {
                    "type": "ECOSYSTEM",
                    "repo": "",
                    "events": [
                        {
                            "introduced": "0",
                            "fixed": "3.7.9-20.oe1",
                            "last_affected": "",
                            "limit": ""
                        }
                    ],
                    "database_specific": null
                }
            ],
            "versions": null,
            "ecosystem_specific": {
                "aarch64": [
                    "python3-devel-3.7.9-20.oe1.aarch64.rpm",
                    "python3-debuginfo-3.7.9-20.oe1.aarch64.rpm",
                    "python3-debugsource-3.7.9-20.oe1.aarch64.rpm",
                    "python3-debug-3.7.9-20.oe1.aarch64.rpm",
                    "python3-3.7.9-20.oe1.aarch64.rpm"
                ],
                "noarch": [
                    "python3-help-3.7.9-20.oe1.noarch.rpm"
                ],
                "src": [
                    "python3-3.7.9-20.oe1.src.rpm"
                ],
                "x86_64": [
                    "python3-debugsource-3.7.9-20.oe1.x86_64.rpm",
                    "python3-devel-3.7.9-20.oe1.x86_64.rpm",
                    "python3-debuginfo-3.7.9-20.oe1.x86_64.rpm",
                    "python3-3.7.9-20.oe1.x86_64.rpm",
                    "python3-debug-3.7.9-20.oe1.x86_64.rpm"
                ]
            },
            "database_specific": null
        },
        {
            "package": {
                "ecosystem": "openEuler-20.03-LTS-SP2",
                "name": "python3",
                "purl": "pkg:rpm/python3@3.7.9-19.oe1"
            },
            "severity": null,
            "ranges": [
                {
                    "type": "ECOSYSTEM",
                    "repo": "",
                    "events": [
                        {
                            "introduced": "0",
                            "fixed": "3.7.9-19.oe1",
                            "last_affected": "",
                            "limit": ""
                        }
                    ],
                    "database_specific": null
                }
            ],
            "versions": null,
            "ecosystem_specific": {
                "aarch64": [
                    "python3-debug-3.7.9-19.oe1.aarch64.rpm",
                    "python3-debuginfo-3.7.9-19.oe1.aarch64.rpm",
                    "python3-devel-3.7.9-19.oe1.aarch64.rpm",
                    "python3-3.7.9-19.oe1.aarch64.rpm",
                    "python3-debugsource-3.7.9-19.oe1.aarch64.rpm"
                ],
                "noarch": [
                    "python3-help-3.7.9-19.oe1.noarch.rpm"
                ],
                "src": [
                    "python3-3.7.9-19.oe1.src.rpm"
                ],
                "x86_64": [
                    "python3-debugsource-3.7.9-19.oe1.x86_64.rpm",
                    "python3-devel-3.7.9-19.oe1.x86_64.rpm",
                    "python3-debuginfo-3.7.9-19.oe1.x86_64.rpm",
                    "python3-3.7.9-19.oe1.x86_64.rpm",
                    "python3-debug-3.7.9-19.oe1.x86_64.rpm"
                ]
            },
            "database_specific": null
        },
        {
            "package": {
                "ecosystem": "openEuler-20.03-LTS-SP3",
                "name": "python3",
                "purl": "pkg:rpm/python3@3.7.9-21.oe1"
            },
            "severity": null,
            "ranges": [
                {
                    "type": "ECOSYSTEM",
                    "repo": "",
                    "events": [
                        {
                            "introduced": "0",
                            "fixed": "3.7.9-21.oe1",
                            "last_affected": "",
                            "limit": ""
                        }
                    ],
                    "database_specific": null
                }
            ],
            "versions": null,
            "ecosystem_specific": {
                "aarch64": [
                    "python3-devel-3.7.9-21.oe1.aarch64.rpm",
                    "python3-3.7.9-21.oe1.aarch64.rpm",
                    "python3-debugsource-3.7.9-21.oe1.aarch64.rpm",
                    "python3-debuginfo-3.7.9-21.oe1.aarch64.rpm",
                    "python3-debug-3.7.9-21.oe1.aarch64.rpm"
                ],
                "noarch": [
                    "python3-help-3.7.9-21.oe1.noarch.rpm"
                ],
                "src": [
                    "python3-3.7.9-21.oe1.src.rpm"
                ],
                "x86_64": [
                    "python3-debugsource-3.7.9-21.oe1.x86_64.rpm",
                    "python3-3.7.9-21.oe1.x86_64.rpm",
                    "python3-debug-3.7.9-21.oe1.x86_64.rpm",
                    "python3-devel-3.7.9-21.oe1.x86_64.rpm",
                    "python3-debuginfo-3.7.9-21.oe1.x86_64.rpm"
                ]
            },
            "database_specific": null
        }
    ],
    "references": [
        {
            "type": "ADVISORY",
            "url": "https://nvd.nist.gov/vuln/detail/CVE-2022-0391"
        },
        {
            "type": "ADVISORY",
            "url": "https://nvd.nist.gov/vuln/detail/CVE-2021-4189"
        }
    ],
    "credits": null,
    "database_specific": {
        "severity": "Medium"
    }
}